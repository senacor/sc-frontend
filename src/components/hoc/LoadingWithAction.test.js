import React from 'react';
import { shallow } from 'enzyme';
import withLoadingAction from './LoadingWithAction';

describe('withLoadingAction Function', () => {
  it('should call the loadingFunction', () => {
    const fn = jest.fn();

    const Wrapped = withLoadingAction(fn)()(() => <div />);

    expect(fn).toHaveBeenCalledTimes(0);
    const component = shallow(<Wrapped />);

    expect(fn).toHaveBeenCalledTimes(1);
    expect(component).toMatchSnapshot();
  });
});

it('sets the display name correctly', () => {
  const TestComponent = () => <div />;
  TestComponent.displayName = 'TestComponent';
  const Wrapped = withLoadingAction()()(TestComponent);

  expect(Wrapped.displayName).toEqual('WithLoadingAction(TestComponent)');
});
