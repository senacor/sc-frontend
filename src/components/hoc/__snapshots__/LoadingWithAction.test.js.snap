// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoadingAction Function should call the loadingFunction 1`] = `
<Connect(LoadingActionRender)
  loadingAction={Array []}
>
  <Component />
</Connect(LoadingActionRender)>
`;
